src/Charcoal/Email/Exception/EmailNotSentException.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Charcoal\Email\Exception;
+
+use Exception;
+
+/**
+ * EmailNotSentException
+ *
+ * Exception thrown when sending an email fails and there is absolutely no doubt no email has been sent.
+ */
+class EmailNotSentException extends Exception
```

Comment:
I would add an interface to easily catch any Charcoal-specific exceptions:

```php
<?php

namespace Charcoal\Email\Exception;

/**
 * Interface for exceptions.
 */
interface ExceptionInterface
{
}
```

I would also make this exception extend `RuntimeException` since its for errors that can only occur on runtime.

```suggestion
class EmailNotSentException extends RuntimeException implements ExceptionInterface
```